fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/MvNormalizePredicateRule.java

```diff
@@ -0,0 +1,127 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Inc.
+
+package com.starrocks.sql.optimizer.rewrite.scalar;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import com.starrocks.analysis.Expr;
+import com.starrocks.catalog.Function;
+import com.starrocks.catalog.FunctionSet;
+import com.starrocks.catalog.Type;
+import com.starrocks.sql.optimizer.Utils;
+import com.starrocks.sql.optimizer.operator.scalar.BinaryPredicateOperator;
+import com.starrocks.sql.optimizer.operator.scalar.BinaryPredicateOperator.BinaryType;
+import com.starrocks.sql.optimizer.operator.scalar.CallOperator;
+import com.starrocks.sql.optimizer.operator.scalar.CompoundPredicateOperator;
+import com.starrocks.sql.optimizer.operator.scalar.ConstantOperator;
+import com.starrocks.sql.optimizer.operator.scalar.ScalarOperator;
+import com.starrocks.sql.optimizer.rewrite.ScalarOperatorRewriteContext;
+
+import java.math.BigInteger;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Set;
+
+public class MvNormalizePredicateRule extends NormalizePredicateRule {
+    // Normalize Binary Predicate
+    // for integer type:
+    // a < 3 => a <= 2
+    // a > 3 => a >= 4
+    // a = 3 => a >= 3 and a <= 3
```

Comment:
why a = 3 => a >= 3 and a <= 3?

Reply:
because I want to normalized the range scalar operator to the same format. Here I choose <= and >=.
I will change <, >, = into <= and >=, then I can judge that a = 3 is contained by a <= 3; a <= 3 is equal to a < 4 for Integer type.